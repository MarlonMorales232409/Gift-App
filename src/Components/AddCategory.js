import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);

    console.log("Handle Change was called");
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((cats) => [...cats, inputValue]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        placeholder='Try a Categorie like "One Punch"'
      />
    </form>
  );
};

export default AddCategory;

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
