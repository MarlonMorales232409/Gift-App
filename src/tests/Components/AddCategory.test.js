import React from "react";
import { shallow } from "enzyme";
import AddCategory from "../../Components/AddCategory";

describe("The component should be show it correctly", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("It should show it corretly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should change the text box", () => {
    const input = wrapper.find("input");
    const value = "Hello World";
    input.simulate("change", { target: { value } });

    expect(wrapper.find("p").text().trim()).toBe(value);
  });

  test("It shouldn't posting the information in submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });
});
