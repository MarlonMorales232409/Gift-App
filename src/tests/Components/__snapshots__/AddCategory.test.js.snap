// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component should be show it correctly It should show it corretly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Try a Categorie like \\"One Punch\\""
    type="text"
    value=""
  />
</form>
`;
